core/trino-main/src/main/java/io/trino/execution/scheduler/ScaledWriterScheduler.java

```diff
@@ -95,12 +98,19 @@ private int getNewTaskCount()
                 .mapToDouble(full -> full ? 1.0 : 0.0)
                 .average().orElse(0.0);
 
-        long writtenBytes = writerTasksProvider.get().stream()
-                .map(TaskStatus::getPhysicalWrittenDataSize)
-                .mapToLong(DataSize::toBytes)
-                .sum();
+        boolean taskWritersLimitExceeded;
+        if (taskScaleWriters) {
+            taskWritersLimitExceeded = writerTasksProvider.get().stream().allMatch(TaskStatus::isWritersFullyScaled);
+        }
+        else {
+            long writtenBytes = writerTasksProvider.get().stream()
```

Comment:
I didn't do it because of two reasons.

1. This change will make other exchangers complicated since we will have to track `physicalWrittenBytes` in those too.
2. In future, we have plans to merge `scale_task_writer` and `scale_writer` into a single config. So, at that time we won't require this extra condition.

Reply:
discussed offline